src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -833,6 +839,39 @@ private static ulong CurlSendCallback(IntPtr buffer, ulong size, ulong nitems, I
                 return Interop.Http.CURL_READFUNC_ABORT;
             }
 
+            private static void SetLowSpeedLimit(EasyRequest easy, TimeSpan timeout)
+            {
+                // If there's no send/receive timeout on the handler, the defaults will
+                // remain disabled, and we don't need to worry about setting or resetting.
+                if (!easy._handler._hasSendReceiveTimeout)
+                {
+                    return;
+                }
+
+                // We need to ensure the timeout is set appropriately, which may include
+                // disabling a timeout currently set if one was.
+
+                long limit = 0, time = 0; // 0 == disable
+
+                if (timeout != Timeout.InfiniteTimeSpan)
+                {
+                    double secs = timeout.TotalSeconds;
+                    if (secs >= 1)
+                    {
+                        limit = 1;
```

Comment:
> Otherwise a Timer or using Wait with a timeout.

See below.

> Yeah, that's another place that feels weird. Since curl is firing the callback after each successfully read header, do we really want 30 seconds per header? If it's just the gap until "the headers" then maybe between each header we'd use the body timeout? (Since we don't know if it was the last header or not) Or use a decaying timer so there's BodyTimeout from "start of first header" to "start of body".

It's not 30 seconds per header.  Setting the speed limit at the beginning of each header callback is no different than setting it once at the beginning, but I do it for each callback because there's no "ok, I'm about to start reading headers" event, nor is there one for redirects getting processed.

I can't tell if you're arguing against the implementation or against the meaning of these timeouts.  If the former, I'd welcome clarification and suggestions.  If the latter, my understanding is that this is what WinHttp exposes, and WinHttpHandler just passes the timeouts along.  If we want to expose something different, then we'd need to do so for the Windows implementation as well, which would mean doing something custom.

We also already have a wealth of different cancellation / timeout options.  Even before this change:
- You can set a timeout on the whole operation via HttpClient.Timeout.
- You can set a timeout on just the connect phase using HttpClientHandler.ConnectTimeout.
- You can pass in your own CancellationToken to SendAsync and cancel it whenever you want, including passing in a CT that has a timer associated with it.
- You can pass a CancellationToken in to each ReadAsync/WriteAsync/FlushAsync operation on the response stream as you're reading from it, if you choose to do so manually.
- You can call HttpClient.CancelAllPending to cancel all pending operations.
- You can Dispose of the response or response stream at any time to cancel the remainder of that operation.

These just add yet another set of fine-grained things to cancel.  I'd be fine if we choose not to expose these, in which case I can just close this PR.  Instead, I think we'd want to implement the ReadTimeout and WriteTimeout properties on the response streams for both the Windows and Unix versions.